packages/sol/src/verification.ts

```diff
@@ -0,0 +1,408 @@
+  const parsedRuleStart = Date.parse(ruleStart);
+  const ruleEndTime = new Date(ruleEnd).getTime();
+  const ruleStartTime = new Date(ruleStart).getTime();
+
+  if (!parsedRuleEnd) return FieldError.ISOInvalidRuleEnd;
+  if (!parsedRuleStart) return FieldError.ISOInvalidRuleStart;
+  if (ruleEndTime < ruleStartTime) return FieldError.RuleEndBeforeStart;
+  if (ruleEndTime < currentTime) return FieldError.RuleEndDatePassed;
+  return true;
+}
+
+export function isValidRuleDuration(
+  duration: number
+): FieldError.InvalidRuleDuration | FieldError.MaxRuleDuratoinExceeded | true {
+  if (duration < MIN_RULE_DURATION_MINUTES)
+    return FieldError.InvalidRuleDuration;
+  if (duration > MAX_RULE_DURATION_MINUTES)
+    return FieldError.MaxRuleDuratoinExceeded;
+  return true;
+}
+
+export function isValidInterviewFeedback(
+  feedback: string
+):
+  | FieldError.TooLongInterviewFeedback
+  | FieldError.TooShortInterviewFeedback
+  | FieldError.InvalidInterviewFeedback
+  | true {
+  const feedbackText = getSafeInnerHtmlText(feedback);
+
+  if (!HTML_REGEX.test(feedback)) return FieldError.InvalidInterviewFeedback;
+  if (feedbackText.length < MIN_FEEDBACK_TEXT_LENGTH)
+    return FieldError.TooShortInterviewFeedback;
+  if (feedbackText.length > MAX_FEEDBACK_TEXT_LENGTH)
+    return FieldError.TooLongInterviewFeedback;
+  return true;
+}
+export function isValidInterviewNote(
+  note: string
+):
+  | FieldError.TooLongInterviewNote
+  | FieldError.TooShortInterviewNote
+  | FieldError.InvalidInterviewNote
+  | true {
+  const noteText = note.replace(HTML_TAGS_REGEX, "");
+
+  if (!HTML_REGEX.test(note)) return FieldError.InvalidInterviewNote;
+  if (noteText.length < MIN_NOTE_TEXT_LENGTH)
+    return FieldError.TooShortInterviewNote;
+  if (noteText.length > MAX_NOTE_TEXT_LENGTH)
+    return FieldError.TooLongInterviewNote;
+  return true;
+}
+
+export function isValidInterviewLevel(
+  level: number
+): FieldError.TooLowInterviewLevel | FieldError.TooHighInterviewLevel | true {
+  if (level < MIN_INTERVIEW_LEVEL) return FieldError.TooLowInterviewLevel;
+  if (level > MAX_INTERVIEW_LEVEL) return FieldError.TooHighInterviewLevel;
+  return true;
+}
+
+export function isValidRatingValue(
+  ratingValue: number
+): FieldError.TooLowRatingValue | FieldError.TooHighRatingValue | true {
+  if (ratingValue < MIN_RATING_VALUE) return FieldError.TooLowRatingValue;
+  if (ratingValue > MAX_RATING_VALUE) return FieldError.TooHighRatingValue;
+  return true;
+}
+
+export function isValidRatingText(
+  ratingText: string
+): FieldError.TooLongRatingText | FieldError.TooShortRatingText | true {
+  if (ratingText.length > MAX_RATING_TEXT_LENGTH)
+    return FieldError.TooLongRatingText;
+  if (ratingText.length < MIN_RATING_TEXT_LENGTH)
+    return FieldError.TooShortRatingText;
+  return true;
+}
+
+export function isValidPlanAlias(
+  planAlias: string
+): FieldError.TooLongPlanAlias | FieldError.TooShortPlanAlias | true {
```

Comment:
The plan alias should be a plain Arabic text. Numbers and especial characters are allowed.  

```suggestion
export function isValidPlanAlias(
  alias: string
): FieldError.LongPlanAlias | FieldError.ShortPlanAlias | true {
```